### DISCUSSION TITLE
[Feature request]: Add Taskbar Segments

### DISCUSSION THREAD

**Discussion by ioanca** [valinet/ExplorerPatcher#3410] [Ideas]
A few months ago I was still using Windows 11 and used to keep my taskbar like this (of course, without the task view and search button - but it's a screenshot i ripped of youtube)
![image](https://github.com/valinet/ExplorerPatcher/assets/98697951/c0a84b73-e112-461c-850c-7d32e2b46587)
"Segmented taskbar" u get it. (from Taskbar XI)
Now I switched back to W10 due to how bad W11 still feels 3 years post-launch.
I use Explorer Patcher mainly for the visuals.
I like my taskbar icons centered with start button and my start menu floating.
But I wanted to have that segmented taskbar so I installed TaskbarX
Sadly, that and ExplorerPatcher don't get quite well together.
The code for segmented taskbar gets confused when confronted with the centered start button resulting in all the icons getting offset to the right part of the screen. So ExplorerPatcher can't segment but can center start button; TaskbarX can segment but can't center start button.
I then tried running RoundedTB. That one didn't have a problem with the centered start button, but it didn't have an automatically resizable segment. You could only resize the segment by dragging the "links" from the taskbar which is kind of a bummer.
RoundedTB said "resizable segments coming soon to windows 10" but that app hasn't got updated in quite a while.
So I was wondering. Would it be that hard to implement such a thing in Explorer Patcher?
Ideally, it would have a setting like "keep only main segment" (which means - make the taskbar visible only under start button and main icons and leave the clock and system tray floating there in the corner like an overlay)
something like this:
![image](https://github.com/valinet/ExplorerPatcher/assets/98697951/890ad22c-48b2-48aa-ba71-6d0d60f59ffc)
and one setting for corner radius.

![image](https://github.com/valinet/ExplorerPatcher/assets/98697951/2e9f5461-ee7d-4b9b-9e54-e6b6b87c19f5)

**Comment by pyrates999** [comment]
Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.
